File: backend/CertificatesManagerApi/Mappers/UserMapper.cs
using CertificatesManagerApi.DTOs;
using CertificatesManagerApi.Entities;

namespace CertificatesManagerApi.Mappers
{
    public class UserMapper
    {
        public static UserDTO ToDto(User user)
        {
            UserDTO userDto = new()
            {
                Handle = user.Handle,
                Name = user.Name,
                FirstName = user.FirstName,
                Email = user.Email,
                UserId = user.UserId,
                Department = user.Department,
                Plant = user.Plant
            };

            return userDto;
        }

        public static User ToEntity(UserDTO userDTO)
        {
            return new()
            {
                Handle = userDTO.Handle,
                Name = userDTO.Name,
                FirstName = userDTO.FirstName,
                Email = userDTO.Email,
                UserId = userDTO.UserId,
                Department = userDTO.Department,
                Plant = userDTO.Plant
            };
        }
    }
}

File: frontend/src/common/utils/mappers/user-mapper.ts
import { UserDTO } from '../../models/dtos/user-dto';
import User from '../../models/user.model';

export class UserMapper {
	static ToModel(userDTO: UserDTO): User {
		let user: User = {
			id: userDTO.handle,
			name: userDTO.name,
			firstName: userDTO.firstName,
			email: userDTO.email,
			userId: userDTO.userId,
			department: userDTO.department,
			plant: userDTO.plant,
		};

		return user;
	}

	static ToDTO(user: User): UserDTO {
		let userDTO: UserDTO = {
			handle: user.id,
			name: user.name,
			firstName: user.firstName,
			email: user.email,
			userId: user.userId,
			department: user.department,
			plant: user.plant,
		};

		return userDTO;
	}
}
